service/opam_build.ml

```diff
@@ -83,3 +90,16 @@ let dockerfile ~base ~info ~repo ~variant =
     ) @@
   copy ~chown:"opam" ~src:["."] ~dst:"/src/" () @@
   run "%s opam install -tv ." caches
+
+let cache = Hashtbl.create 10000
```

Comment:
Why not just make this `7` or something small instead and let it size up?  It'll keep the default memory profile of ocurrent a bit lower (not relevant for Docker builds I realise, but possibly if used in other contexts)

Reply:
This is in ocaml-ci, not OCurrent, so it won't affect anything else.